### PR TITLE
feat(CapMan): More `tenant_ids` for TSDB Queries

### PR DESCRIPTION
**Pull request by rahul-kumar-saini**
### Overview
- Added `organization_id` to more Snuba Requests, specifically targeting `tsdb-modelid:4` queries as they are the biggest chunk of queries Snuba is handling in production
- Most calls to `tsdb.get_range` handled as well, some I wasn't able to figure out
- More context: #44788

### CODE REVIEW

File: src/sentry/api/serializers/models/group_stream.py [hunk 1]
```diff
@@ -153,11 +153,13 @@ def query_tsdb(self, groups: Sequence[Group], query_params, **kwargs):
         except Environment.DoesNotExist:
             stats = {g.id: tsdb.make_series(0, **query_params) for g in groups}
         else:
+            org_id = groups[0].project.organization_id if groups else None
```
Comment (rahul-kumar-saini) [inline src/sentry/api/serializers/models/group_stream.py:156]: Assumes all groups in the list belong to a single org, let me know if this assumption is incorrect.

Comment (JoshFerge) [reply src/sentry/api/serializers/models/group_stream.py:156]: that assumption seems correct. is `groups` guaranteed to have at least one element though?

**Review by JoshFerge — CHANGES_REQUESTED** [review]
all look good to me except for `group_stream`.

File: src/sentry/api/serializers/models/group_stream.py (hunk 1 above)
Comment (JoshFerge) [reply src/sentry/api/serializers/models/group_stream.py:156]: ah right, `if groups` will evaluate to `False` if it is empty.

**Review by JoshFerge — APPROVED** [review]

**Pull request merged**